Source/FullScreenPokemon.ts

```diff
@@ -6517,29 +6517,96 @@ module FullScreenPokemon {
-                defenderName: string = args.defenderName,
-                attacker: IThing = <IThing>FSP.BattleMover.getThing(attackerName),
-                defender: IThing = <IThing>FSP.BattleMover.getThing(defenderName),
-                direction: number = attackerName === "player" ? 1 : -1,
-                notes: IThing[] = [
+            let attackerName: string = args.attackerName;
+            let defenderName: string = args.defenderName;
+            let attacker: IThing = <IThing>FSP.BattleMover.getThing(attackerName);
+            let defender: IThing = <IThing>FSP.BattleMover.getThing(defenderName);
+            let direction: number = attackerName === "player" ? 1 : -1;
+            let notes: IThing[] = [
                     FSP.ObjectMaker.make("Note"),
                     FSP.ObjectMaker.make("Note")
                 ];
+            let menu: IMenu = <IMenu>FSP.MenuGrapher.getMenu("BattleDisplayInitial");
+            let dt: number = 10;
+            let flip1: number = 1;
+            let flip2: number = 1;
+            let startX: number;
+            let startY: number;
+            let differenceX: number;
+            let differenceY: number;
+
+            if (direction === 1) {
+                startX = menu.left + attacker.width / 2 * FSP.unitsize;
+                startY = menu.bottom - attacker.height * FSP.unitsize;
+                differenceX = menu.right - startX;
+                differenceY = (menu.top + defender.height / 2 * FSP.unitsize) - startY;
+            } else {
+                startX = menu.right - attacker.width / 2 * FSP.unitsize;
+                startY = menu.top + attacker.height * FSP.unitsize;
+                differenceX = menu.left - startX;
+                differenceY = (menu.bottom - defender.height * FSP.unitsize) - startY;
+            }
+
+            FSP.addThing(notes[0], startX, startY);
+            FSP.TimeHandler.addEvent(
+                FSP.addThing,
+                2,
+                notes[1],
+                startX + notes[1].width / 2 * FSP.unitsize,
+                startY + FSP.unitsize * 3);
+            FSP.TimeHandler.addEventInterval(
+                function (note: IThing): void {
+                    setTimeout(
+                        function (): void {
+                            FSP.shiftHoriz(note, differenceX / 4);
+                            if (flip1 === 1) {
+                                FSP.shiftVert(note, differenceY / 10 * 6);
+                            } else {
+                                FSP.shiftVert(note, -1 * differenceY / 8);
+                            }
+                            flip1 = flip1 * -1;
+                        },
+                        dt);
+                },
+                dt,
+                4,
+                notes[0]);
+            FSP.TimeHandler.addEventInterval(
+                function (note: IThing): void {
+                    setTimeout(
```

Comment:
Don't use `setTimeout`; always prefer `TimeHandler`. `setTimeout` and `setInterval` don't respect game pausing or framerate changing.